[OpenSprinkler] binding throwing class cast exception

It appears that the OpenSprinkler binding has a bug in it. I see the following exception:

```
2019-12-21 20:48:29.644 [ERROR] [nal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.handleCommand()' on 'org.openhab.binding.opensprinkler.internal.handler.OpenSprinklerStationHandler@2c64d7': org.eclipse.smarthome.core.library.types.QuantityType cannot be cast to org.eclipse.smarthome.core.library.types.DecimalType

java.lang.ClassCastException: org.eclipse.smarthome.core.library.types.QuantityType cannot be cast to org.eclipse.smarthome.core.library.types.DecimalType

	at org.openhab.binding.opensprinkler.internal.handler.OpenSprinklerStationHandler.handleNextDurationCommand(OpenSprinklerStationHandler.java:92) ~[?:?]

	at org.openhab.binding.opensprinkler.internal.handler.OpenSprinklerStationHandler.handleCommand(OpenSprinklerStationHandler.java:76) ~[?:?]
```

If you look at [this method](https://github.com/openhab/openhab-addons/blob/master/bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/handler/OpenSprinklerStationHandler.java#L86), there is a bug. Firstly, the method is the following:

```java
    private void handleNextDurationCommand(ChannelUID channelUID, Command command) {
        if (!(command instanceof QuantityType<?>)) {
            logger.info("Ignoring implausible non-DecimalType command for NEXT_DURATION");
            return;
        }
        this.nextDurationTime = ((QuantityType<?>) command).toBigDecimal();
        updateState(channelUID, (DecimalType) command);
    }
```

In this code we are checking if the `command` is a `QuantityType`. If it is not, we fail. Therefore, after this first 'if' statement, we know that command is a `QuantityType`. The next line gets the duration time as a big decimal. The final line is type-casting the `command` argument, which we know is a `QuantityType` to be a `DecimalType`, which results in the class path exception. 

// cc @FlorianSW

--- a/bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/handler/OpenSprinklerStationHandler.java
+++ b/bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/handler/OpenSprinklerStationHandler.java
@@ -85,11 +85,12 @@ public class OpenSprinklerStationHandler extends OpenSprinklerBaseHandler {
 
     private void handleNextDurationCommand(ChannelUID channelUID, Command command) {
         if (!(command instanceof QuantityType<?>)) {
-            logger.info("Ignoring implausible non-DecimalType command for NEXT_DURATION");
+            logger.info("Ignoring implausible non-QuantityType command for NEXT_DURATION");
             return;
         }
-        this.nextDurationTime = ((QuantityType<?>) command).toBigDecimal();
-        updateState(channelUID, (DecimalType) command);
+        QuantityType<?> quantity = (QuantityType<?>) command;
+        this.nextDurationTime = quantity.toBigDecimal();
+        updateState(channelUID, quantity);
     }
 
     private void handleStationStateCommand(OpenSprinklerApi api, Command command) {

